### PR TITLE
fix: loading service from an absolute path

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## What does this PR address?

This is the case when running `Service.serve_http()`

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/_bentoml_impl/loader.py [hunk 1]
```diff
@@ -70,20 +71,23 @@ def normalize_identifier(
...
-            return normalize_package(service_identifier), pathlib.Path(
-                working_dir if working_dir is not None else "."
-            )
+            module_spec, _, service_path = service_identifier.partition(":")
+            module_path = pathlib.Path(module_spec)
+            if not module_path.suffix:
+                return service_identifier, pathlib.Path(working_dir or ".")
+            if not module_path.is_absolute():
+                module_path = pathlib.Path(working_dir or ".").joinpath(module_path)
+            return f"{module_path.stem}:{service_path}", module_path.parent
```
Comment (bojiang) [inline src/_bentoml_impl/loader.py:88]: Shall we return CWD as the bento path?

Comment (frostming) [reply src/_bentoml_impl/loader.py:88]: bento_path here is to tell `import_service` where to import the service, so it makes sense to return the parent dir of it.

This is to handle the case where `bento_identifier` is a path.

Comment (bojiang) [reply src/_bentoml_impl/loader.py:88]: bento_path is used to be the working directory of sub-processes

Comment (frostming) [reply src/_bentoml_impl/loader.py:88]: So `normalize_identifier` is to return two values, import string and bento_path. For a bento identifier as absolute path. What shall it return? `normalize_identifier("/path/to/package.py:Inference")`

Comment (bojiang) [reply src/_bentoml_impl/loader.py:88]: @frostming

Comment (bojiang) [reply src/_bentoml_impl/loader.py:88]: You just said this case happens when user call `service.serve_http()`, thus I think we should inherit the working directory, which is '.'
Will that work?
@frostming

Comment (frostming) [reply src/_bentoml_impl/loader.py:88]: but the first return value will be directly feed to `import_service`. An absolute path isn't a valid import string right?

**Pull request merged**